src/attrib.d

```diff
@@ -411,6 +411,11 @@ public:
     {
         assert(msg);
         char* depmsg = null;
+        sc = sc.startCTFE();
+        this.msg = this.msg.semantic(sc);
+        this.msg = resolveProperties(sc, this.msg);
+        sc = sc.endCTFE();
+        this.msg = this.msg.optimize(WANTexpand);
```

Comment:
To evaluate the `msg` expression, `setScope` is too early. For example, it doesn't work properly for:

```d
enum a1 = "a", b1 = "b";
deprecated(a1 ~ b1) int v1 = 1; // OK

deprecated(a2 ~ b2) int v2 = 2; // NG
enum a2 = "a", b2 = "b";
```

UDA doesn't have same issue. In order to avoid forward reference issues, UDA is evaluated at when `__traits(getAttribute)` is requested (it happens in `semantic` stage at the earliest), or done in `UserAttributeDeclaration.semantic2` stage at the latest.